internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -659,6 +661,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			id = strconv.Itoa(i)
 		}
 
+		if !cd.Synced {
+			log.Debug("Composed resource is not yet synced", "id", id)
+			unsynced = append(unsynced, cd)
+			r.record.Event(xr, event.Normal(reasonCompose, fmt.Sprintf("Composed resource %q is not yet synced", id)))
```

Comment:
I wonder if we could just set `Ready = false` when we can't apply a composed resource, rather than introducing a new Synced field? I'm not convinced there's a significant value in making the distinction.

The reason we're not just returning early when we hit an invalid composed resource is because we think the composed resource might eventually become valid. If that is the case, I think it's fair to think of it as "not ready yet".

On the other hand it's possible the composed resource could just stay invalid forever until a human intervenes. If that is the case, maybe "ready" would be misleading.

Reply:
Right now we set it to synced false in case of errors, and so we did before 1.14. It makes sense to me to have it still be marked as ready but possibly not synced, it means it's still healthy, just not updated to the latest desired state for some reason.

Reply:
`Composed resource is not yet synced` was to mirror `Composed resource is not yet ready`.
But I agree it could be misleading, to avoid being too specific risking breaking this assumption later on, I'd rather continue talking about it being synced or not. The whole point is that we couldn't update/create a composed resource according to the current desired state for the composite. Maybe `Couldn't sync Composed resources`?

Reply:
I think we may be talking past each other - maybe we can sync on this live?

I think we have different understanding of what "synced" means.

Reply:
@phisco and I just chatted on Zoom.

We'll change the wording in the error string to say something like applied instead of synced, to avoid confusion with the composed resource's synced status condition.

The `cd.Synced` struct field is less of an issue. I could see a case for it being `Synced` there (after all `Ready` already doesn't _strictly_ mean the composed resource's `Ready` status condition, but instead our general assessment of whether it's ready). I could go either way on that.